backend/open_api_spec.yml

```diff
@@ -0,0 +1,683 @@
+                        contact_email:
+                          type: string
+                        curator_name:
+                          type: string
+                        revised_at:
+                          type: number
+                        created_at:
+                          type: number
+                        links:
+                          $ref: "#/components/schemas/links"
+                        datasets:
+                          type: array
+                          items:
+                            $ref: "#/components/schemas/dataset_preview"
+                        revision_of:
+                          type: string
+                          description: uuid of the public version of this Collection
+                        access_type:
+                          type: string
+                          enum: [ READ, WRITE ]
+                        published_at:
+                          type: number
+                        description:
+                          type: string
+                        obfuscated_uuid:
+                          type: string
+                        data_submission_policy_version:
+                          type: string
+                        genesets:
+                          type: array
+                          items:
+                            $ref: "#/components/schemas/gene_set"
+        "401":
+          $ref: "#/components/responses/401"
+    post:
+      description: Create a new Collection
+      security:
+        - cxguserCookie: []
+      requestBody:
+        content:
+          application/json:
+            $ref: "#/components/schemas/collection_form_metadata"
+      responses:
+        "201":
+          description: Created
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/collection_uuid"
+        "400":
+          $ref: "#/components/responses/400_multiple"
+        "401":
+          $ref: "#/components/responses/401"
+
+  /curation/collections/{collection_id}:
+    delete:
+      description: Delete a private Collection. Delete an existing revision for a public Collection.
+      security:
+        - cxguserCookie: []
+      parameters:
+        - $ref: "#/components/parameters/path_collection_uuid"
+      responses:
+        "202":
+          description: Accepted
+          content:
+            application/json:
+              type: string
+        "401":
+          $ref: "#/components/responses/401"
+    get:
+      description: Fetch Collection metadata and associated Datasets, public and private.
+      security:
+        - cxgUserCookieLenient: []
+      parameters:
+        - $ref: "#/components/parameters/path_collection_uuid"
+        - $ref: "#/components/parameters/query_visibility"
+      responses:
+        "200":
+          description: OK
+          content:
+            application/json:
+              schema:
+                allOf:
```

Comment:
This seems wrong. This should return the same response as the `GET /curation/collections` but for a singe collection.

Reply:
Just using some inheritance concepts. The `collection_form_metadata` is the subset of collection data that is user-provided. The stuff returned from this collection endpoint is 1) all of that, plus 2) the fields that get added (published_at, revised_at, datasets).

Reply:
Although I suppose that, since the submission form does not necessarily mean we would want to return that field for the collections endpoint, perhaps I should split this back out. Just because it happens to line up doesn't mean it makes logical sense. I assume that's your criticism.